### PR TITLE
Enh: Consolidate `(is|set)(Database)Installed()` to `ApplicationTrait`

### PR DESCRIPTION
**Pull request by martin-rueegg**
- Code de-duplication
- move code away from `Settings`, that has nothing to do with `Settings`
- move it to `BaseSettingsManager` where also `isDatabaseInstalled()` resides

### What kind of change does this PR introduce?

- Refactor

### Does this PR introduce a breaking change?

- No

### The PR fulfills these requirements

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [x] All tests are passing
- [ ] New/updated tests are included
- [x] Changelog was modified

**Other information:**

### CODE REVIEW

**Comment by martin-rueegg** [comment]
All failing errors seem to stem from
- #6714

File: MIGRATE-DEV.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Version 1.16 (Unreleased)
 - `\humhub\modules\content\components\ContentAddonActiveRecord::canWrite()`
 - `\humhub\modules\file\models\File::canRead()` use `canView()` instead
 - `\humhub\modules\content\components\ContentAddonActiveRecord::canRead()` use `canView()` instead
+- `\humhub\models\Setting::isInstalled()` use `BaseSettingsManager::isInstalled()` instead
```
Comment (luke-) [inline MIGRATE-DEV.md:14]: I like the improvement. Just as a thought, could this also be something for `ApplicationTrait`?  

e.g. `Yii::$app->isInstalled()`. 


```php
    public static function isInstalled(): bool
    {
        return isset($this->params['installed']) && $this->params['installed'];
    }

```

**Review by luke- — CHANGES_REQUESTED** [review]

File: MIGRATE-DEV.md (hunk 1 above)
Comment (martin-rueegg) [reply MIGRATE-DEV.md:14]: Yes, I guess that would be an even better solution! (Although the method would then have to be non-static)

If we do this, I'd like to also move the `BaseSettingsManager::isDatabaseInstalled()` there. And maybe even the `DynamicSetting:set(Database)Installed()`, too.

Would you agree?

Comment (luke-) [reply MIGRATE-DEV.md:14]: Hmmm, good point.

Yes, then let's move all 4 methods `is|set[Database]Installed()` to `ApplicationTrait`. Seems to be the best place.

Comment (martin-rueegg) [reply MIGRATE-DEV.md:14]: on it!

**Review by luke- — APPROVED** [review]

**Comment by luke-** [comment]
@martin-rueegg It's much better that way! Thanks for the refactoring!

**Pull request merged**